operations/image_management/Upload_and_Register_an_Image_Recipe.md

```diff
@@ -131,3 +131,145 @@ Download and expand recipe archives from S3 and IMS. Modify and upload a recipe
+   NOTE: Only files listed under the template_files key will be templated by IMS. 
+
+2. For each indicated file above, add appropriate Jinja2 variables where required to enable the result that are sought.
+   In the example below, the Jinja2 variable `{{ CSM_RELEASE_VERSION }}` is used to help identify the fully qualified 
+   Nexus repo name/path to use when building the recipe.
+
+   ```bash
+   ncn# cat config.xml
+   ...
+   <repository type="rpm-md" alias="csm-{{ CSM_RELEASE_VERSION }}-sle-15sp3" priority="2" imageinclude="true">
+       <source path="https://packages.local/repository/csm-{{ CSM_RELEASE_VERSION }}-sle-15sp3/"/>
+   </repository>
+   ...
+   ```
+
+   NOTE: The repo referenced above is for documentation purposes only and may not actually exist. 
+
+3. Create a tgz archive of the image recipe.
+
+    ```bash
+    ncn# export ARTIFACT_FILENAME=recipe.tar.gz
+    ncn# tar cvfz ../$ARTIFACT_FILE .
+    ncn# cd ..
+    ```
+
+### Add template key/value pairs to IMS recipe record
+
+1. Create a new IMS recipe record with template_dictionary key/value pairs.
+
+    ```bash
+    ncn# cray ims recipes create --name "My Recipe" \
+    --recipe-type kiwi-ng --linux-distribution sles15 \
+    --template-dictionary-key CSM_RELEASE_VERSION \
+    --template-dictionary-value 1.2.5
+    
+    created = "2018-12-04T17:25:52.482514+00:00"
+    id = "2233c82a-5081-4f67-bec4-4b59a60017a6"
+    linux_distribution = "sles15"
+    name = "my_recipe.tgz"
+    recipe_type = "kiwi-ng"
+    
+    [[template_dictionary]]
+    key = "CSM_RELEASE_VERSION"
+    value = "1.2.5"
+    ```
+
+    Additional key/value pairs can be added by providing a list of comma-separated keys/values to the 
+    --template-dictionary-key and --template-dictionary-value parameters.  
+
+    If successful, create a variable for the `id` value in the returned data.
+
+    ```bash
+    ncn# export IMS_RECIPE_ID=2233c82a-5081-4f67-bec4-4b59a60017a6
+    ```
+   
+2. Upload the customized recipe to S3.
+
+   It is suggested as a best practice that the S3 object name start with `recipes/` and contain the IMS recipe ID to remove ambiguity.
+
+   ```bash
+   ncn# cray artifacts create ims recipes/$IMS_RECIPE_ID/$ARTIFACT_FILENAME $ARTIFACT_FILENAME
+   ```
+
+3. Update the IMS recipe record with the S3 path to the recipe archive.
+
+   ```bash
+   ncn# cray ims recipes update $IMS_RECIPE_ID \
+   --link-type s3 \
+   --link-path s3://ims/recipes/$IMS_RECIPE_ID/$ARTIFACT_FILENAME
+
+   id = "2233c82a-5081-4f67-bec4-4b59a60017a6"
+   recipe_type = "kiwi-ng"
+   linux_distribution = "sles15"
+   name = "my_recipe.tgz"
+   created = "2020-02-05T19:24:22.621448+00:00"
+
+   [[template_dictionary]]
+   key = "CSM_RELEASE_VERSION"
+   value = "1.2.5"
+   
+   [link]
+   path = "s3://ims/recipes/2233c82a-5081-4f67-bec4-4b59a60017a6/recipe.tar.gz"
+   etag = ""
+   type = "s3"
+   ```
+
+### Building an image from an IMS templated recipe
```

Comment:
Change to:

Build an Image from an IMS Templated Recipe

Reply:
Done